src/App.tsx

```diff
@@ -15,48 +16,128 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortBy {
+  Alphabet = 'alphabet',
+  Length = 'length',
+  Default = 'Default',
+}
+
+const sortAlphabet = (a: string, b: string, direction: boolean) => {
+  return direction
+    ? a.localeCompare(b)
+    : b.localeCompare(a);
+};
+
+const sortByLenght = (a: string, b: string, direction: boolean) => {
+  if (a.length === b.length) {
+    return sortAlphabet(a, b, direction);
+  }
+
+  return direction
+    ? a.length - b.length
+    : b.length - a.length;
+};
+
+function getSortGoods(
+  goods: string[],
+  sortField: SortBy,
+  sortDirection: boolean,
+) {
+  const preparedGoods: string[] = [...goods];
+
+  if (sortField) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SortBy.Alphabet:
+          return sortAlphabet(good1, good2, sortDirection);
+
+        case SortBy.Length:
+          return sortByLenght(good1, good2, sortDirection);
+
+        case SortBy.Default:
+          return !sortDirection
+            ? preparedGoods.indexOf(good2) - preparedGoods.indexOf(good1)
+            : 0;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  return preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState(SortBy.Default);
+  const [isReversed, setIsReversed] = useState(true);
+  const sortGoods = getSortGoods(goodsFromServer, sortField, isReversed);
+  const sortReverse = () => {
+    if (isReversed) {
+      setIsReversed(false);
+    } else {
+      setIsReversed(true);
```

Comment:
```suggestion
    setIsReversed(prevValue => !prevValue);
```
It is better to use functional state here with toggling the previous value